AdNetworkSupport/MoPubMediationLogger.m

```diff
@@ -0,0 +1,124 @@
+//
+// Created by Pooja Shashidhar on 4/16/18.
+// Copyright (c) 2018 MoPub. All rights reserved.
+//
+#import <Foundation/Foundation.h>
+#import <UIKit/UIKit.h>
+#import "MoPubMediationLogger.h"
+#import "MPLogging.h"
+
+
+@interface MoPubMediationLogger ()
+@property(nonatomic) NSString *NETWORK_TYPE;
+@property(nonatomic) NSString *ADFORMAT_TYPE;
+@property(nonatomic) Event EVENT_TYPE;
+
+@end
+
+@implementation MoPubMediationLogger
+
+
+NSDictionary *eventDictionary = nil;
+NSDictionary *networkDictionary = nil;
+NSDictionary *adFormatDictionary = nil;
+
+NSArray *adFormatArray = nil;
+
+// Type A implementation: Fetch network name and ad format directly from network adapter
+// Simple and elegant solution as the details are fetched during network intialization. Adaption of the code is also made easy
+-(instancetype)initWithNetworkType: (Network)networkType  AndAdFormat:(AdFormat)adFormat
+{
+
+    self = [super init];
+    if(self)
+    {
+        self.NETWORK_TYPE=networkDictionary[@(networkType)];
+        self.ADFORMAT_TYPE=adFormatDictionary[@(adFormat)];
+        
+    }
+    return self;
+}
+
+//Type B implementation: Fetch network name and ad format from parsing the class names from the adapter
+// This process of fetching data is cumbersome and requires to build parsing logic
+-(instancetype)initWithClassName:(NSString *)className
+{
+    
+    self = [super init];
+
+    if(self)
+    {
+        [MoPubMediationLogger load];
+        [self parseSdkAndAdFormat:className];
+    }
+    return self;
+}
+
+-(void)parseSdkAndAdFormat:(NSString *)className
+{
+    for(NSString * item in adFormatArray){
+        NSArray *substrings = [className componentsSeparatedByString:item];
+        if(substrings.count==2){
+            self.NETWORK_TYPE= [substrings objectAtIndex:0];
+            self.ADFORMAT_TYPE=item;
+            break;
+        }
+    }
+    if(!self.NETWORK_TYPE){
+          self.NETWORK_TYPE = @"Unknown Network";
+    }
+    if(!self.ADFORMAT_TYPE){
+        self.ADFORMAT_TYPE = @"Unknown Adformat ";
+    }
+}
+
+//Logger method to log the details such as network name, ad format and callback information
+- (void)log :(Event) eventKey
+{
+    MPLogInfo(@"%@-%@-%@", self.NETWORK_TYPE,self.ADFORMAT_TYPE,eventDictionary[@(eventKey)]);
+    
+}
+
++ (void) load
+{
+    eventDictionary = @{
```

Comment:
Not all of these events are used in your demo (like `AD_REQUESTED`). I'd add as many as applicable.